ldms/src/core/ldms_xprt.c

```diff
@@ -1723,8 +1727,11 @@ static int __process_dir_set_info(struct ldms_set *lset, enum ldms_dir_type type
 		}
 		pair = nxt_pair;
 	}
+err:
 	if (lset) {
 		pthread_mutex_unlock(&lset->lock);
+		if (rc)
```

Comment:
@baallan I think this is not correct because if `lset` is NULL, the function will always return 0, regardless of the error. Consider the `goto err;` on line 1697. We should separate the error out and the normal out.

```
out:
    if (lset) {
        pthread_mutex_unlock(&lset->lock);
        if (type == ....) {
          ....
        }
    }
    return 0;
err:
    if (lset)
        pthread_mutex_unlock(&lset->lock);
    return rc;
```

Reply:
@nichamon you are correct that in the lset==null case, the rc gets lost. We can keep simplify and keep the out-path code unified by simply changing the final 'return 0' to 'return rc'. I've pushed a revision that does this.